commonknowledge/meep-intelligence-hub
# Add posthog, pageviews and some example custom events [MEEP-197]

## Description
This PR 
- installs Posthog in the NextJs app and links it to Common Knowledge Posthog account and Common Knowledge website project
- adds a pageview wrapper function around the whole app so we can see page views
- adds a utlity function for creating custom events and a few examples of this around things like data source creation and report creation
Also added Meep frontend url to Session replay Authorized domains for replay so we can record user sessions

It looks like we already have button click tracking in the project.
Let me know if you'd like to add data attributes to specific buttons or any other custom events.

## Motivation and Context
Addresses issue [MEEP-197](https://linear.app/commonknowledge/issue/MEEP-197/install-posthog-product-analytics)

## How Can It Be Tested?
Download the branch and run locally
Add Posthog API key and host to .env file
Follow the user flow to submit some events to Posthog (e.g. view a page; try to create a data source, map report etc)
Observe the events in Posthog dashboard

## How Will This Be Deployed?
Posthog environment variables will need to be added to Render frontend

## Types of changes
